diff.py

```diff
@@ -32,7 +32,7 @@ def compare_source_files_objects(obj1, obj2, ignore_hits):
                         if func1['exec'] is False and func2['exec'] is True:
                             diff_funcs.append(func2)
 
-    if len(diff_funcs) == 0 and len(diff_cov) == 0:
+    if len(diff_funcs) == 0 and len(diff_cov) == 0 or all(cov == 0 or cov is None for cov in diff_cov) and len(diff_funcs) == 0:
```

Comment:
If you rewrite the condition as:
```Python
if len(diff_funcs) == 0 and (len(diff_cov) == 0 or all(cov == 0 or cov is None for cov in diff_cov))
```

it will be less confusing.

Even better, the `len(diff_cov) == 0` case is already covered by `all(cov == 0 or cov is None for cov in diff_cov)` (as if diff_cov is [], then the `all` will return True).

So you can do:
```Python
if len(diff_funcs) == 0 and all(cov == 0 or cov is None for cov in diff_cov):
```

Reply:
Please also add a test that covers this case

Reply:
There was a case in diff report that looked like: `{"coverage": [long list with nulls and 0s], "branches": [], "functions": [{"name": "_ZNK7mozilla13DefaultDeleteIA_cEclEPc.isra.201", "start": 539, "exec": true}, {"name": "_ZN7mozilla9UniquePtrIA_cNS_13DefaultDeleteIS1_EEE5resetEDn.isra.202", "start": 482, "exec": true}]`. Seems, it will be lost if I remove a check for len(diff_funcs).

Reply:
Ok, clear. :)